src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -1023,7 +1023,11 @@ module T = struct
       Or_error.try_with (fun () ->
           List.map cs ~f:(fun cmd ->
               let open Ledger in
-              User_command.to_verifiable ~ledger ~get ~location_of_account cmd
+              User_command.to_verifiable
+                ~find_vk:
+                  (Zkapp_command.Verifiable.find_vk_via_ledger ~ledger ~get
```

Comment:
Won't this get the wrong vk if one zkApp command sets it and a later one attempts to use it? It feels like we need to accumulate across all of the commands.